File: airbyte-commons/src/main/java/io/airbyte/commons/util/AutoCloseableIterator.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.util;

import java.util.Iterator;

/**
 * If you operate on this iterator, you better close it. {@link AutoCloseableIterator#close} must be
 * idempotent. The contract on this interface is that it may be called MANY times.
 *
 * @param <T> type
 */
public interface AutoCloseableIterator<T> extends Iterator<T>, AutoCloseable {}

File: airbyte-cdk/bulk/toolkits/legacy-source-integration-tests/src/testFixtures/kotlin/io/airbyte/cdk/test/fixtures/legacy/AutoCloseableIterator.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.test.fixtures.legacy

/**
 * If you operate on this iterator, you better close it. [AutoCloseableIterator.close] must be
 * idempotent. The contract on this interface is that it may be called MANY times.
 *
 * @param <T> type </T>
 */
interface AutoCloseableIterator<T> : MutableIterator<T>, AutoCloseable, AirbyteStreamAware
